Objects/listobject.c

```diff
@@ -1362,9 +1362,9 @@ _list_extend(PyListObject *self, PyObject *iterable)
         Py_END_CRITICAL_SECTION2();
     }
     else {
-        Py_BEGIN_CRITICAL_SECTION(self);
+        Py_BEGIN_CRITICAL_SECTION2(self, iterable);
```

Comment:
I don't think we should be locking `iterable` here. If I understand correctly `iterable` is a generator. We need to enforce that only one thread can be executing a generator at a time, but that logic should be in `gen_send_ex2`.

Reply:
I thought about that too, yeah. Would you prefer the list change be removed entirely?

Reply:
I've updated this to add a lock to native generators instead of in `list.__init__`. I'm still a tiny bit concerned about not locking here, but I guess I'll address that in another PR if it's problematic.

Reply:
Ken already has a PR open related to this: https://github.com/python/cpython/pull/120327

I am not in a rush to fix this crash, given that it's only come up from fuzzing, and would rather we are careful not to regress generator performance.

Reply:
Oh, we already have nearly the exact same fix :laughing: I'll close this in favor of that PR.